From b267bb3d8e466243dc2dfbfc04563ef97309cb45 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Fri, 12 Oct 2012 14:36:36 +0800
Subject: [PATCH 350/498] ENGR00227841 wifi:rewrite compat-wireless's compile

Back to use absolute path for cross compile tool.
If possible, can change to use relative path together with kernel
and u-boot. Only change one place won't work.

Change-Id: I98aae4fd179c16b0de79f63ad9de3b961dd87253
Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 wlan/atheros/compat-wireless/Android.mk                      | 5 +++--
 wlan/atheros/compat-wireless/include/linux/compat_autoconf.h | 1 -
 wlan/atheros/compat-wireless/scripts/gen-compat-autoconf.sh  | 3 +--
 3 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/wlan/atheros/compat-wireless/Android.mk b/wlan/atheros/compat-wireless/Android.mk
index fb8be9e..8a625d1 100644
--- a/wlan/atheros/compat-wireless/Android.mk
+++ b/wlan/atheros/compat-wireless/Android.mk
@@ -32,18 +32,19 @@ ATH_ANDROID_SRC_BASE:= $(LOCAL_PATH)
 ATH_ANDROID_ROOT:= $(CURDIR)
 
 ATH_LINUXPATH=$(ATH_ANDROID_ROOT)/kernel_imx
+ATH_CROSS_COMPILE=$(ATH_ANDROID_ROOT)/prebuilts/gcc/linux-x86/arm/arm-eabi-4.6/bin/arm-eabi-
 
 mod_cleanup := $(ATH_ANDROID_ROOT)/$(ATH_ANDROID_SRC_BASE)/dummy
 
 $(mod_cleanup) :
-	$(MAKE) -C $(ATH_ANDROID_SRC_BASE) ARCH=arm CROSS_COMPILE=$(ARM_EABI_TOOLCHAIN)/arm-eabi- KLIB=$(AT\
+	$(MAKE) -C $(ATH_ANDROID_SRC_BASE) ARCH=arm CROSS_COMPILE=$(ATH_CROSS_COMPILE) KLIB=$(AT\
   H_LINUXPATH) KLIB_BUILD=$(ATH_LINUXPATH) clean
 	mkdir -p $(TARGET_OUT)/etc/firmware/ath6k/AR6003/hw2.1.1/
 	mkdir -p $(TARGET_OUT)/lib/modules/
 
 ath6kl_module_file :=drivers/net/wireless/ath/ath6kl/ath6kl_sdio.ko
 $(ATH_ANDROID_SRC_BASE)/$(ath6kl_module_file):$(mod_cleanup) $(TARGET_PREBUILT_KERNEL) $(ACP)
-	$(MAKE) -C $(ATH_ANDROID_SRC_BASE) O=$(ATH_LINUXPATH) ARCH=arm CROSS_COMPILE=$(ARM_EABI_TOOLCHAIN)/arm-eabi- KLIB=$(ATH_LINUXPATH) KLIB_BUILD=$(ATH_LINUXPATH)
+	$(MAKE) -C $(ATH_ANDROID_SRC_BASE) O=$(ATH_LINUXPATH) ARCH=arm CROSS_COMPILE=$(ATH_CROSS_COMPILE) KLIB=$(ATH_LINUXPATH) KLIB_BUILD=$(ATH_LINUXPATH)
 	$(ACP) -fpt $(ATH_ANDROID_SRC_BASE)/compat/compat.ko $(TARGET_OUT)/lib/modules/
 	$(ACP) -fpt $(ATH_ANDROID_SRC_BASE)/net/wireless/cfg80211.ko $(TARGET_OUT)/lib/modules/
 
diff --git a/wlan/atheros/compat-wireless/include/linux/compat_autoconf.h b/wlan/atheros/compat-wireless/include/linux/compat_autoconf.h
index a7d6487..6e227ea 100644
--- a/wlan/atheros/compat-wireless/include/linux/compat_autoconf.h
+++ b/wlan/atheros/compat-wireless/include/linux/compat_autoconf.h
@@ -2,7 +2,6 @@
 #define COMPAT_AUTOCONF_INCLUDED
 /*
  * Automatically generated C config: don't edit
- * Sat Sep 29 15:20:31 CST 2012 
  * compat-wireless-2.6: patch-10-22-gc53d81a-p
  * linux-2.6: msm-2012-06-21-43-g7ffaacb
  */
diff --git a/wlan/atheros/compat-wireless/scripts/gen-compat-autoconf.sh b/wlan/atheros/compat-wireless/scripts/gen-compat-autoconf.sh
index 28ed25a..145a458 100755
--- a/wlan/atheros/compat-wireless/scripts/gen-compat-autoconf.sh
+++ b/wlan/atheros/compat-wireless/scripts/gen-compat-autoconf.sh
@@ -1,6 +1,7 @@
 #!/bin/bash
 # 
 # Copyright 2007	Luis R. Rodriguez <mcgrof@winlab.rutgers.edu>
+# Copyright (C) 2011-2012 Freescale Semiconductor, Inc.
 #
 # Use this to parse a small .config equivalent looking file to generate
 # our own autoconf.h. This file has defines for each config option
@@ -35,7 +36,6 @@ fi
 
 CREL=$(cat $COMPAT_RELEASE | tail -1)
 KREL=$(cat $KERNEL_RELEASE | tail -1)
-DATE=$(date)
 
 # Defines a CONFIG_ option if not defined yet, this helps respect
 # linux/autoconf.h 
@@ -133,7 +133,6 @@ cat <<EOF
 #define COMPAT_AUTOCONF_INCLUDED
 /*
  * Automatically generated C config: don't edit
- * $DATE 
  * compat-wireless-2.6: $CREL
  * linux-2.6: $KREL
  */
-- 
1.8.0

